### DISCUSSION TITLE
Faceted search can be improved

### DISCUSSION THREAD

**Discussion by stereobooster** [oramasearch/orama#537] [Ideas]
Faceted search can be improved using the same approach as [ItemsJS](https://github.com/itemsapi/itemsjs) (e.g. [FastBitSet.js](https://github.com/lemire/FastBitSet.js/)).

Compare:

- https://faceted.stereobooster.com/pages/itemsjs/
- https://faceted.stereobooster.com/pages/orama/

And article about comparison: https://stereobooster.com/posts/faceted-search/

**Comment by SergeyRe** [comment]
i think first point to improve  in your both examples is generate initial state on site generation step
check this approach here    [nobelists]( https://www.nobelists.org/en/all/) (itemsjs used)
also you can add generation of persistent index and restoring it on  page load (for orama case)
i think that orama search on ready index is enough fast
